airflow/api_fastapi/execution_api/routes/variables.py

```diff
@@ -67,6 +67,20 @@ def get_variable(variable_key: str, token: deps.TokenDep) -> VariableResponse:
     return VariableResponse(key=variable_key, value=variable_value)
 
 
+@router.post(
+    "",
+    responses={
+        status.HTTP_401_UNAUTHORIZED: {"description": "Unauthorized"},
+        status.HTTP_403_FORBIDDEN: {"description": "Task does not have access to the variable"},
+    },
+)
+def post_variable(body: VariablePostBody, token: deps.TokenDep) -> VariableResponse:
+    """Set an Airflow Variable."""
+    # TODO: add the token validation
```

Comment:
I am yet to handle this part, wasn't too sure if I should follow the current template or ignore for now

Reply:
Call `has_variable_access`, and add a new parameter to that function to make it `has_variable_access(variable_key, token, write=True)` etc.

Though we should probably make better use of decorators/dependencies so we don't need to handle this _inside_ each view function, but for now just follow the pattern.

Reply:
Sure, did this: 
```
    if not has_variable_access(body.key, token, write_access=True):
        raise HTTPException(
            status_code=status.HTTP_403_FORBIDDEN,
            detail={
                "reason": "access_denied",
                "message": f"Task does not have access to write variable {body.key}",
            },
        )

```